internal/clients/http/azure/create_vm.go

```diff
@@ -0,0 +1,395 @@
+// strongly inspired by https://github.com/Azure-Samples/azure-sdk-for-go-samples/tree/main/sdk/resourcemanager/compute/create_vm
+// from commit 2ea51d9744d5d4680b4983b2e292a7b1d0085ff4
+
+package azure
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
+	"github.com/RHEnVision/provisioning-backend/internal/ptr"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/codes"
+)
+
+var subscriptionId string
+
+const TraceName = "github.com/RHEnVision/provisioning-backend/internal/clients/http/azure"
+
+const (
+	resourceGroupName = "redhat-deployed2"
+	vmName            = "redhat-vm"
+	vnetName          = "redhat-vnet"
+	subnetName        = "redhat-subnet"
+	nsgName           = "redhat-nsg"
+	nicName           = "redhat-nic"
+	diskName          = "redhat-disk"
+	publicIPName      = "redhat-public-ip"
+	location          = "eastus"
+	adminUsername     = "azureuser"
+)
+
+func (c *client) CreateVM(ctx context.Context, imageID string, sshKeyBody string) (*armcompute.VirtualMachine, error) {
+	ctx, span := otel.Tracer(TraceName).Start(ctx, "CreateVM")
+	defer span.End()
+
+	logger := logger(ctx)
+	logger.Debug().Msg("Creating Azure VM instance")
+
+	resourceGroup, err := c.createResourceGroup(ctx, resourceGroupName, location)
+	if err != nil {
+		span.SetStatus(codes.Error, "cannot create resource group")
+		logger.Error().Err(err).Msgf("cannot create resource group")
```

Comment:
Idea: This could easily be a `ctxval.LogErrorWithSpan(ctx, span, err, "msg")` and that function would:

* Pull the logger from the context
* Set span status
* Log the error

I really appreciate amount of logging and tracing! Well done.